stelaval/assignment1/Anagram.java

```diff
@@ -0,0 +1,95 @@
+package assignment1;
+
+import java.util.Arrays;
+
+public class Anagram {
+
+	/**
+	 * Determines if one string is an anagram of another string
+	 * 
+	 * @param s1 first string
+	 * 
+	 * @param s2 second string
+	 * 
+	 * @return true if s1 and s2 are anagrams of each other
+	 */
+	static boolean isAnagram(String s1, String s2) {
+
+		String string1 = s1.replaceAll("\\s", "");
+		String string2 = s2.replaceAll("\\s", "");
+
+		if (string1.length() != string2.length()) {
+			return false;
+		} else {
+
+			char[] s1Array = string1.toCharArray();
+			char[] s2Array = string2.toCharArray();
+
+			Arrays.sort(s1Array);
+			Arrays.sort(s2Array);
+
+			return Arrays.equals(s1Array, s2Array);
+		}
+	}
+
+	/**
+	 * Handles both case sensitive and case insensitive anagrams
+	 */
+	
+	static boolean isWordAnagram(String s1, String s2, boolean isCaseSensitive) {
+		if (isCaseSensitive) {
+			return isAnagram(s1, s2);
+		} else {
+			return isAnagram(s1.toLowerCase(), s2.toLowerCase());
+		}
+	}
+
+	/**
+	 * Determines if one sentence is an anagram of another sentence
+	 * 
+	 * @param s1 first sentence
+	 * 
+	 * @param s2 second sentence
+	 * 
+	 * @return true if s1 and s2 are anagrams of each other
+	 */
+	static boolean isSentenceAnagram(String s1, String s2) {
+
+		String[] sentence1 = s1.split("\\s*");
+		String[] sentence2 = s2.split("\\s*");
+
+		if (sentence1.length != sentence2.length) {
+			return false;
+		} else {
+
+			boolean[] isAnagramFound = new boolean[sentence1.length];
+
+			for (int i = 0; i < sentence1.length; i++) {
+
+				for (int j = 0; j < sentence2.length; j++) {

```

Comment:
This algorithm works if one takes assignment literally - sentences are anagrams if every word from first one is an anagram of any word from the second. This means that "abc abc" and "abc def" are anagrams.

What if the definition was narrowed down so that each word from each sentence can be used only once, i.e. "abc abc" and "abc def" are not anagrams, but "abc def" and "abc fed" are. And similarly "abc abc" and "abc abc abc" were not anagram.

How would that change algorithm?

Reply:
I will change it